### PR TITLE
Search: add search form

### LINKED ISSUE #35: Hotels search form
Please create hotels search form. For this, you should modify and reuse `DatePicker` and `TextInput` from `@kiwicom/native-common` package. Do just simple form (location API is not ready, guest popup and the final design will be in next issue). Now it's important to have good component design and proper form behavior. Also please connect it to the component variables created in #34.

zpl://screen?sid=5a177284d9724be52c416a4d&pid=5a1308cdae8ae89cae5c15af

### PR DESCRIPTION
**Pull request by jakubzaba**
Closes #35

### CODE REVIEW

**Comment by mrtnzlml** [comment]
Please test these changes in [Expo](https://docs.expo.io/versions/latest/introduction/installation.html#mobile-client-expo-for-ios-and-android) after build has finished:

  QR Code

File: package.json [hunk 1]
```diff
@@ -16,6 +16,8 @@
   },
   "dependencies": {
     "expo": "^23.0.0",
+    "lodash": "^4.17.4",
+    "moment": "^2.19.4",
```
Comment (mrtnzlml) [inline package.json]: This is not how monorepo works. You need to update `package.json` in the "hotels" package. Yarn even warn you about this...

File: app/core/src/navigation/ExploreStackNavigator.js [hunk 2]
```diff
@@ -23,6 +23,7 @@ export default StackNavigator(
   },
   {
     ...NavigatorOptions,
-    initialRouteName: 'Home',
+    // initialRouteName: 'Home',
+    initialRouteName: 'AllHotels',
```
Comment (mrtnzlml) [inline app/core/src/navigation/ExploreStackNavigator.js]: This is a mistake.

File: app/hotels/src/allHotels/AllHotels.js [hunk 3]
```diff
@@ -2,49 +2,99 @@
...
+        checkin: search.checkin,
+        checkout: search.checkout,
+        roomsConfiguration: search.roomsConfiguration,
+      },
+    });
+  };
+
+  isReadyToSearch = () => {
+    const { search } = this.state;
+    return search.checkin && search.checkout;
```
Comment (mrtnzlml) [inline app/hotels/src/allHotels/AllHotels.js]: I think the location is important as well.

File: app/hotels/src/allHotels/AllHotels.js [hunk 4]
```diff
@@ -2,49 +2,99 @@
...
-        force: true,
-      }}
-    />
-  );
+  render = () => {
+    const { search } = this.state;
+    return (
+      <View style={styles.wrapper}>
+        <SearchForm onChange={this._handleSearchChange} />
+        {this.isReadyToSearch() && (
```
Comment (mrtnzlml) [inline app/hotels/src/allHotels/AllHotels.js]: I think some kind of default search would be better than the empty page.

File: app/hotels/src/allHotels/SearchForm.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    flex: 1,
+    borderRadius: 0,
+  },
+  buttonText: {
+    color: Color.grey._900,
+  },
+});
+
+type Props = {
+  onChange: (search: Search) => void,
```
Comment (mrtnzlml) [inline app/hotels/src/allHotels/SearchForm.js]: I think it should be possible to use exact types here (same for the state).

File: app/hotels/src/allHotels/SearchForm.js [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    this._handleOnChange();
+  };
+
+  _handleCheckinChange = date => this._handleDateChange(date, 'checkin');
+
+  _handleCheckoutChange = date => this._handleDateChange(date, 'checkout');
+
+  _handleDateChange = async (date, type) => {
+    await this.setState({ [type]: date });
+    this._handleOnChange();
```
Comment (mrtnzlml) [inline app/hotels/src/allHotels/SearchForm.js]: I don't think the `await this.setState` works. I mean it works but not as you expect. You should use second parameter of the function in this case. Something like this:

```js
this.setState({ [type]: date }, () => this._handleOnChange());
```

Method `setState` [doesn't return `Promise`](https://github.com/facebook/react/blob/master/packages/react/src/ReactBaseClasses.js#L53). Correct me if I am wrong.

File: app/hotels/src/allHotels/AllHotels.js [hunk 7]
```diff
@@ -2,49 +2,99 @@
...
+              }
+            `}
+            variables={{
+              search: {
+                ...search,
+                checkin: search.checkin
+                  ? moment(search.checkin, 'DD/MM/YYYY').format('YYYY-MM-DD')
+                  : null,
+                checkout: search.checkout
+                  ? moment(search.checkout, 'DD/MM/YYYY').format('YYYY-MM-DD')
```
Comment (mrtnzlml) [inline app/hotels/src/allHotels/AllHotels.js]: Do we need the second parameter here? It seems that checkin and checkout dates are actually `Date` type so we are not parsing it here...

Comment (jakubzaba) [reply app/hotels/src/allHotels/AllHotels.js]: Right, it is needed, but not here.

File: app/hotels/src/allHotels/AllHotels.js (hunk 4 above)
Comment (jakubzaba) [reply app/hotels/src/allHotels/AllHotels.js]: Added default search.

**Comment by mrtnzlml** [comment]
Integrated into master. Thanks.

**Pull request closed without merging**